src/app/common/httpclient.ts

```diff
@@ -83,7 +80,18 @@ export class HttpClient {
     }
 
     public request(url: string, options?: RequestOptionsArgs, warn?: boolean): Promise<Response> {
-        return this.requestOnConnection(url, options, warn).toPromise()
+        return new Promise((resolve, reject) => {
+            let sub: Subscription = this.requestOnConnection(url, options, warn).subscribe(
+                v => resolve(v),
+                e => {
+                    let err: any = this.runtime.HandleConnectError(e);
+                    if (err) {
+                        reject(err);
+                    }
+                },
+                () => sub.unsubscribe(),
```

Comment:
The variable "sub" here is local variable. So, this shouldn't be used inside of the complete callback function because these are called asynchronously. 

Please see this article for the detail.

https://www.pluralsight.com/guides/javascript-callbacks-variable-scope-problem

If we really want to unsubscribe, I think the variable should be saved in the parent scope.

Reply:
Even though let has some magic to make the variable in the upper layer, I don't think it is a good practice to save the subscription object variable inside a callback function and make it not readable.

Reply:
This is not the only place where this kind of logic exists. https://github.com/Microsoft/IIS.WebManager/blob/dev/src/app/webserver/webserver.service.ts#L67
I thought we have agreed offline to address this kind of coding pattern issue later?
I disagree with the readability though. The variable is used after it was declared. What is unreadable about this?

Reply:
Yeah, I agreed about that yesterday but I thought we have still the time to fix the issue instead of postponing. But I am okay to investigate this issue in a separate work item.
I opened this issue and assigned it to you. 
https://github.com/Microsoft/IIS.WebManager/issues/336

In my experience, using local variable in callback function is problematic but should be avoided. But if that was used, I have to figure out why that was used and that makes me to read the code difficult. Maybe we can discuss about this in the team meeting. :-)

Reply:
Removed unsubscribe method